### ISSUE TITLE
LaTeX kernel date shows in certain type of test files

### ISSUE THREAD

**Issue by FrankMittelbach** [latex3/l3build#302]
With the 2023-06-01 release LaTeX repeats the kernel date info at the end of the run.
In certain type of test files this shows in the test files generating a diff whenever the format date change, e.g.
```
  Don't change this file in any respect.
  [1
  ] (github-010.aux)
+  ***********
+ LaTeX2e <2023-06-01> patch level 1
+ L3 programming layer <2023-06-30>
+  ***********
```
I think we should normalize/drop these lines  in the test output to avoid unnecessary fails on regular basis.

Maybe the simplest solution would be
```
\RemoveFromHook{enddocument/info}[kernel/release]
```
inside regression-test.tex.

**Comment by josephwright** [comment]
If you are dropping the hook, for 2e that would be done 'locally' (in `test2e`): is that what you mean?

**Comment by FrankMittelbach** [comment]
yes test2e.tex and regression-test.tex I think, maybe guared to do it only if running on top of the LaTeX kernel in case of regression-test.tex

**Comment by FrankMittelbach** [comment]
I came across it when rerunning the test for footmisc which has tests that include `\end{document}` and uses `regression-test.tex`.

**Comment by davidcarlisle** [comment]
why did we add this to the log?

**Comment by josephwright** [comment]
@davidcarlisle  Karl asked us, I think - it was because it can get lost at the start, depending on how you start your TeX run

**Comment by u-fischer** [comment]
@davidcarlisle  https://github.com/latex3/latex2e/issues/944

**Comment by muzimuzhi** [comment]
In resolving latex3/latex2e#944, [some code](https://github.com/latex3/latex2e/blob/a03efc85ea25975405982d869a0d636f984fedbb/support/regression-test.tex#L298C1-L314C4) (also pasted below) which ignores the added log lines, is added (see latex3/latex2e#950) to latex2e's own `regression-test.tex`. Maybe it's time to move it to l3build's `regression-test.tex`, the "source".

```tex
%%% this needs to be moved to the source of regression-test.tex!

\ifx\AddToHook\undefined
\else

%
% We don't want release information showing up in \end{document}
% because that would change over and over again.
% We therefore drop that information from the hook (should also
% happen in the standard regression test when running a LaTeX
% format). By using a voids relation we can reactivate the code in
% a test by dropping "the testmode" label.
%
\AddToHook{enddocument/info}[kernel/testmode]{}
\DeclareHookRule{enddocument/info}{kernel/testmode}{voids}{kernel/release}

\fi
```

**Comment by muzimuzhi** [comment]
> I came across it when rerunning the test for footmisc which has tests that include `\end{document}` and uses `regression-test.tex`.

> _Update:_ Wait a minute. Looks like such code is already added, see [07ce1dd](https://github.com/latex3/l3build/commit/07ce1dda7bf2c1f13260e0061ba01ee8abda4f68) which closed #255. Then why the log lines are not suppressed?

I cloned `footmisc` repo and checked out commit just prior to commit FrankMittelbach/fmitex-footmisc@fac18b7 (drop kernel info in tests, 2023-07-05), ran `l3build check` and can't reproduce the problem.  LaTeX kernel date written to log at the end of document were automatically suppressed.

```
$ git clone --depth=1 git@github.com:FrankMittelbach/fmitex-footmisc.git ltx-pkg-footmisc
$ cd ltx-pkg-footmist
$ git fetch --depth=1 origin 7a79984
$ git switch --detach 7a79984
$ l3build check
```

I did get two failed tests, but none of the cause is related to the added LaTeX kernel date. It seems the difference for `github-001.pdftex.diff` came from different versions of `lipsum` package.

```
  Check failed with difference files
  - ./build/test/github-001.pdftex.diff
  - ./build/test/github-010.pdftex.diff
```

**Comment by FrankMittelbach** [comment]
It seems I had a local version of regression-test.tex still in the path ... thanks for checking
